kubejs/client_scripts/item_modifiers/jei_descriptions.js

```diff
@@ -429,82 +429,61 @@ onEvent('jei.information', (event) => {
         {
             items: ['simplefarming:brewing_barrel'],
             text: [`${id_prefix}brewing_barrel_2`]
+        },
+        {
+            items: disabledItems,
+            text: [`${id_prefix}disabled_please_report`]
+        },
+        {
+            items: generatableCobblestone,
+            text: [`${id_prefix}generatable_cobble`]
+        },
+        {
+            items: generatableStone,
+            text: [`${id_prefix}generatable_stone`]
+        },
+        {
+            items: refinedStorageItems.map((item) => `refinedstorage:${item}`),
+            text: [`${id_prefix}can_be_dyed`]
+        },
+        {
+            items: [
+                'compact_drawer',
+                'drawer_controller',
+                'slave',
+                'trim',
+                'full_one',
+                'full_two',
+                'full_four',
+                'half_one',
+                'half_two',
+                'half_four'
+            ].map((item) => 'framedcompactdrawers:framed_' + item),
+            text: [`${id_prefix}framed_drawers`]
         }
     ];
 
-    const simplefarming_beverages = [
-        { type: 'beer', ingredient: 'Barley', effect: 'Strength' },
-        { type: 'cauim', ingredient: 'Cassava', effect: 'Jump Boost' },
-        { type: 'cider', ingredient: 'Apples', effect: 'Luck' },
-        { type: 'mead', ingredient: 'Honeycomb', effect: 'Haste' },
-        { type: 'sake', ingredient: 'Rice', effect: 'Speed' },
-        { type: 'tiswin', ingredient: 'Cactus Fruit', effect: 'a Health Boost' },
-        { type: 'vodka', ingredient: 'Potatoes', effect: 'Resistance' },
-        { type: 'whiskey', ingredient: 'Wheat', effect: 'Absorption' },
-        { type: 'wine', ingredient: 'Grapes', effect: 'Regeneration' }
-    ];
+    const simplefarming_beverages = ['beer', 'cauim', 'cider', 'mead', 'sake', 'tiswin', 'vodka', 'whiskey', 'wine'];
 
     simplefarming_beverages.forEach((beverage) => {
         recipes.push({
-            items: [`simplefarming:${beverage.type}`],
-            text: [
-                `Brewed by right-clicking ${beverage.ingredient} into a Brewing Barrel.`,
-                `Grants the imbiber ${beverage.effect}.`
-            ]
+            items: [`simplefarming:${beverage}`],
+            text: [`${id_prefix}beverages`],
+            with: [[`${id_prefix}beverages.${beverage}.ingredient`, `${id_prefix}beverages.${beverage}.effect`]]
         });
     });
 
     recipes.forEach((recipe) => {
         for (let i = 0; i < recipe.text.length; i++) {
-            recipe.text[i] = Text.translate(recipe.text[i]);
+            if (recipe.with[i]) {
+                recipe.with[i] = recipe.with[i].map((str) => Text.translate(str));
+                recipe.text[i] = Text.translate(recipe.text[i], ...recipe.with[i]);
+            } else {
+                recipe.text[i] = Text.translate(recipe.text[i]);
+            }
```

Comment:
Could you please explain what's going on here? :D

Reply:
Sure. Line479 and line480 are two major additions, so I will explain more on these two. 
It will first try to determine if there is a `recipe.with[i]` , which should provides texts for `recipe.text[i]` to replace %s . If not, it will simply use old handlers(line482) .
If there is a `recipe.with[i]` , it will first try to localize texts inside `recipe.with[i]` using `.map((str) => Text.translate(str))` , raw texts will be retained as-is. 
Then those texts will be "unpacked" and passed to translate() function. The `...` in `...recipe.with[i]` is used for unpacking elements. Let's take an example:
Assume an array `arr = ["str1", 2]` and a function `func()` . Calling `func(arr)` equals to `func(["str1", 2])` , while `func(...arr)` equals to `func("str1", 2)` , note that there is no "[" and "]" when using "..." .

Reply:
I see, thank you!
What happens if %s is first in the string? Does the ordering of parameters in `Text.translate(recipe.text[i], ...recipe.with[i]);` matter?

Cause it seems to me like this would produce original string + any occurrences of %s, afterwards?

Reply:
Yes, the ordering of parameters does matter, because translate() function works like this: 
```
function translate(text, ...args){
    t = langFile[text] ? langFile[text] : text.copy()
    args.foreach(arg => {
        t.replaceOnce("%s", arg)
    })
    return t
}
```
(These codes are simplified, not actual source code.)
It will takes the first parameter it received as the base, and other parameters are used for replacing %s . 
That's why recipe.text[i] should be the first parameter, we wants it to be the base. 
And, about `...args` , here it means every parameters that are not the first param, will be packed into an array called "args" . So when passing `…recipe.with[i]` , we needs to unpack first, to avoid nested array which cannot be handled by `t.replaceOnce("%s", arg)` .

Reply:
Ahh okay, thanks for bearing with me and explaining 😃